flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -859,6 +849,14 @@ private void cleanUpRemainingJobData(JobID jobId, boolean jobGraphRemoved) {
             }
         }
         blobServer.cleanupJob(jobId, jobGraphRemoved);
+
+        if (jobGraphRemoved) {
```

Comment:
I just realized that the condition is still invalid. We only want to mark the job as cleaned if all the related artifacts are cleaned. Therefore, we shouldn't just check for the JobGraph but also all the other cleanup steps.

Reply:
👍

Reply:
Ah yeah, of course :facepalm: